cheatsheets_draft/Secrets_Management_CheatSheet.md

```diff
@@ -401,33 +401,97 @@ EaaS is a model in which users subscribe to a cloud-based encryption service wit
 - Key handling and cryptographic implementations are taken care of by Encryption Service, not by developers
 - The provider could add more services to interact with the sensitive data
 
-## 9 Workflow in Case of Compromise
-
-(by @thatsjet)
-
-### 9.1 Process
-
-## 11 Secret detection
-
-(by @thatsjet)
-
-- Many native integrations possible (Cloud platforms, CI/CD tooling, application libraries, container orchestrators)
-- Secret lifecycle (rotation, deletion, lifespan)
-- Key material management (keys to the kingdom)
-- Open source? (Depending on security posture)
-- Encryption (at rest, in transit)
-- Access control (fine-grained)
-- Performance
-- Audit logs
-- Scalable (enterprise)
-- Manageable operations (upgrading, recovery)
-- Agnostic
-- Support for many secrets backends: database, certificates, ssh keys, cloud providers, key/value, etc
-- Dynamic secrets
-- Encryption as a service
-- Fine-grained policies (MFA requirements)
-- Extensibility
-- Documentation
+## 8 Detection
+
+There are many approaches to secrets detection and some very useful open source projects to help with this. Of note, the [Yelp Detect Secrets](https://github.com/Yelp/detect-secrets) project is mature and has signature matching for around 20 secrets.
+
+### 8.1 General detection approaches
+
+- Create standard test secrets and use them universally across the organization. This allows for the reduction of false-positives by only needing to track a single test secret for each secret type.
+- Consider enabling secrets detection at the developer level to avoid checking secrets into code before commit/PR.
+- Make secrets detection part of the threat model. Consider secrets as part of the attack surface during threat modeling exercises.
+- Evaluate detection utilities and related signatures often to ensure they meet expectations.
+- Consider having more than one detection utility and correlating/de-duping results to identify potential areas of detection weakness.
+
+
+### 8.2 Types of secrets to be detected
+
+Many types of secrets exists and signatures for each should be considered to ensure accurate detection for all. Among the more common types are:
+
+- High availability secrets (Tokens that are difficult to rotate)
+- Application configuration files
+- Connection strings
+- Credentials
+- Platform specific secret types (e.g.; Amazon Web Services, Google Cloud)
+
+### 8.3 Detection lifecycle
+
+Secrets are like any other authorization token. They should:
+
+- Exist only for as long as necessary (rotate often)
+- Have a method for automatic rotation
+- Only be visible to those who need them (least privilege)
+- Be revokable (including the logging of attempt to use a revoked secret)
+- Never be logged
+
+Detections should exist during each of these stages of the secret lifecycle to ensure secrets were not introduced but missed in a prior step.
```

Comment:
detection duing logging will mean that double-encoding and/or other things might need to be in palce as well?

Reply:
and/or encryption